libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/MapboxCarNavigationManager.kt

```diff
@@ -42,19 +44,26 @@ class MapboxCarNavigationManager internal constructor(
         onUpdateTrip(trip)
     }
 
-    private val tripSessionStateObserver = TripSessionStateObserver { tripSessionState ->
-        logAndroidAuto("$LOG_CATEGORY tripSessionStateObserver state: $tripSessionState")
-        when (tripSessionState) {
-            TripSessionState.STARTED -> navigationManager.navigationStarted()
-            TripSessionState.STOPPED -> navigationManager.navigationEnded()
+    private var inActiveNavigation = false
+
+    private val routesObserver = RoutesObserver {
+        if (it.navigationRoutes.isEmpty()) {
+            logAndroidAuto("$LOG_CATEGORY stop active navigation")
+            inActiveNavigation = false
+            navigationManager.navigationEnded()
+            MapboxScreenManager.replaceTop(MapboxScreen.FREE_DRIVE)
```

Comment:
@kmadsen @Zayankovsky maybe you know a better place for this?

Reply:
This is the right place to put it :+1:

Reply:
It seems that this was only added because you replaced `stopTripSession()` with `setNavigationRoutes(emptyList())`. Shouldn't this line be placed in `NavigationManagerCallback.onStopNavigation` in this case? In other scenarios someone may want to open a different screen when clearing routes (e.g. if they allow going from active navigation to route preview).

Reply:
I mean it is fine to always go to FreeDrive in this particular case, but in other scenarios it is up to each screen to decide where to go next after clearing routes. For example this is what `ArrivalScreen` does:
https://github.com/mapbox/mapbox-navigation-android/blob/09acc9a23c76115f38e51e725da5f7f327ad09f1/libnavui-androidauto/src/main/java/com/mapbox/androidauto/screenmanager/factories/ArrivalScreenFactory.kt#L24-L25
Developers can override default screens and implement a different logic.

Reply:
Yeah, OK, changed that.